src/live_restore/live_restore_fs.c

```diff
@@ -1581,6 +1589,58 @@ __validate_live_restore_path(WT_FILE_SYSTEM *fs, WT_SESSION_IMPL *session, const
     return (fh->close(fh, (WT_SESSION *)session));
 }
 
+/*
+ * __wt_live_restore_fs_log_copy --
+ *     Copy the log files from the source to the destination prior to recovery. We also need to copy
+ *     over prep log files as logging will expect these are available to be used. This needs to
+ *     happen after __wt_logmgr_config to ensure the relevant logging configuration has been parsed.
+ */
+int
+__wt_live_restore_fs_log_copy(WT_SESSION_IMPL *session)
+{
+    WT_CONNECTION_IMPL *conn = S2C(session);
+    WT_DECL_RET;
+    if (!F_ISSET(&conn->log_mgr, WT_LOG_CONFIG_ENABLED))
+        return (0);
+
+    u_int logcount = 0;
+    char **logfiles = NULL;
+    WT_DECL_ITEM(filename);
+    WT_FH *fh = NULL;
+    uint32_t lognum;
+
+    /* Get a list of actual log files. */
+    WT_ERR(__wt_log_get_files(session, WT_LOG_FILENAME, &logfiles, &logcount));
+    for (u_int i = 0; i < logcount; i++) {
+        WT_ERR(__wt_log_extract_lognum(session, logfiles[i], &lognum));
+        /* Call log open file to generate the full path to the log file. */
+        WT_ERR(__wt_log_openfile(session, lognum, 0, &fh));
+        ret = __wti_live_restore_fs_fill_holes(fh->handle, (WT_SESSION *)session);
```

Comment:
I purposefully opted to not use the `__wt_copy_and_sync` function as it does not take a `read_size` otherwise it is effectively the same. I can try and modify this to use it or spin off another ticket to implement that.

Reply:
Since the `read_size` is best effort anyway, maybe using `wt_copy_and_sync` is good enough. It also reminds me that this code, as written, does not `sync` the log files to the destination disk.

That led me to looking at the normal copy path. Does the `wti_live_restore_fs_fill_holes` want to sync the content it copied to the destination once it is done?

Reply:
I'll update copy_and_sync to take an optional read_size param. I think fsyncing at the end of fill holes prior to exiting the worker thread makes correctness more obvious. I think would be correct in any case, if we crashed after filling the holes in a file but that never actually reached the file the holes would still exist and therefore we'd repeat filling those same holes.

Reply:
I remember now, there's a separate reason we can't use `copy_and_sync`. It is trying to copy a file from `a` to `b` but we're copying it from `a` to `a'`. The live restore fill holes copy is aware of layers and can utilize that but anything above it is not. We could try to be fancy about which paths we give to `copy_and_sync` but it'd break some assumptions.

Reply:
Added a comment: [96881ba](https://github.com/wiredtiger/wiredtiger/pull/11430/commits/96881ba1e912e6311bf550f99c3a2adf7555313a)